dts/bindings/counter/nxp,mcux-pit.yaml

```diff
@@ -3,9 +3,9 @@
 
 description: NXP MCUX Periodic Interrupt Timer (PIT)
 
-compatible: "nxp,kinetis-pit"
+compatible: "nxp,mcux-pit"
```

Comment:
This would break backward compatibility. Please document in the migration guide.

Reply:
> This would break backward compatibility. Please document in the migration guide.

yes, and all the other changes in this PR break backward compatibility, so we should update the migration guide anyways @EmilioCBen

Reply:
> if we are going to rename this binding I'd suggest `nxp,pit`, because MCUX is making reference to an SDK, not hw related. And also to mention it in release notes so downstream users are aware of this change.

One thing I was considering when I changed the binding is that
IF the IMX boards and now with my awareness of the S32 boards support the pit 
THEN should this be considered a kinetis pit suggesting that only works for the kinetis.
However another thing to consider is that currently this is not bare metal and still
uses the APIs from the MCUX SDK so I changed it to mcux,pit.

Do you think I should still change it to nxp,pit?

Reply:
I will update the migration guide in the next push to this PR.

Reply:
I will also update the ST Boards

Reply:
> IMO it should be nxp,pit. The binding is a description of a device. There could be different driver implementations (baremtal, SDKs, etc) for the same binding.

I agree here. We (hopefully) won't end up with multiple different driver implementations, but there is no need to include the software HAL name in the binding compatible. Devicetree is fundamentally intended to describe hardware, and the hardware here is an instance of the PIT IP.

Reply:
I updated the binding to be nxp,pit
I have updated the s32 boards
I have updated the migration guide.

Reply:
@manuargue we don't have your boards so you will need to test for us.